### DISCUSSION TITLE
Mounting ENDF and PENDF files for an NJOY run

### DISCUSSION THREAD

**Discussion by eitan-weinstein** [njoy/NJOY2016#339] [Q&A] [answered]
Hello,

I am trying to use the NJOY GROUPR to create groupwise ENDF files based on the TENDL-2017 Nuclear Data Library, which only has pointwise data. The issue that I am running into is in actually mounting my ENDF and PENDF files in the input script and having read the user manual, I am still unsure how to do this. I imagine I'm just overlooking something simple, but I am new to using NJOY, so I would appreciate some guidance for this.

Here is my input file that I am using:

```
groupr
 20 21 0 31/
 2631 17 0 11 0 1 1 1/
 "GROUPR Processing for Fe56"/
 293
 0/
 3 1 "(n, total)"/
 3 2 "(z,z0)"/
 3 3 "(z, nonelas.)"/
 3 4 "(z,n)"/
 3 5 "(z,anything)"/
 3 11 "(z,2nd)"/
 3 16 "(z,2n)"/
 3 17 "(z,3n)"/
 3 22 "(z,nα)"/
 3 24 "(z,2nα)"/
 3 28 "(z,np)"/
 3 29 "(z,n2α)"/
 3 32 "(z,nd)"/
 3 33 "(z,nt) "/
 3 34 "(z,n3He)"/
 3 41 "(z,2np)"/
 3 42 "(z,3np)"/
 3 44 "(z,n2p)"/
 3 45 "(z,npα)"/
 3 51 "(z,n1)"/
 3 52 "(z,n2)"/
 3 53 "(z,n3)"/
 3 54 "(z,n4)"/
 3 55 "(z,n5)"/
 3 56 "(z,n6)"/
 3 57 "(z,n7)"/
 3 58 "(z,n8)"/
 3 59 "(z,n9)"/
 3 60 "(z,n10)"/
 3 61 "(z,n11)"/
 3 62 "(z,n12)"/
 3 63 "(z,n13)"/
 3 64 "(z,n14)"/
 3 65 "(z,n15)"/
 3 66 "(z,n16)"/
 3 67 "(z,n17)"/
 3 68 "(z,n18)"/
 3 69 "(z,n19)"/
 3 70 "(z,n20)"/
 3 71 "(z,n21)"/
 3 72 "(z,n22)"/
 3 73 "(z,n23)"/
 3 74 "(z,n24)"/
 3 75 "(z,n25)"/
 3 76 "(z,n26)"/
 3 77 "(z,n27)"/
 3 78 "(z,n28)"/
 3 79 "(z,n29)"/
 3 80 "(z,n30)"/
 3 91 "(z,nc)"/
 3 102 "(z,γ)"/
 3 103 "(z,p)"/
 3 104 "(z,d)"/
 3 105 "(z,t)"/
 3 106 "(z,3He)"/
 3 107 "(z,α)"/
 3 108 "(z,2α)"/
 3 111 "(z,2p)"/
 3 112 "(z,pα)"/
 3 115 "(z,pd)"/
 3 116 "(z,pt)"/
 3 117 "(z,dα)"/
 0/
 0/
```

I am executing this through a bash script. When I run the script, I get the Fortran runtime error that it cannot open file 'tape20', which seems sensible to me given that I am not sure how the actual tape20 is connected to my ENDF file itself.

```
#!/bin/bash

# Check if NJOY is installed and accessible
if ! command -v njoy &> /dev/null
then
    echo "NJOY could not be found. Please make sure it is installed and added to your PATH."
    exit 1
fi

# Define the input files
TAPE20="tendl_2017_Fe056_endf.endf"
TAPE21="tendl_2017_Fe056_pendf.pendf"
INPUT="groupr.inp"
OUTPUT="groupr.out"

# Check if input files exist
if [ ! -f "$TAPE20" ]; then
    echo "ENDF file not found!"
    exit 1
fi

if [ ! -f "$TAPE21" ]; then
    echo "PENDF file not found!"
    exit 1
fi

if [ ! -f "$INPUT" ]; then
    echo "Input file not found!"
    exit 1
fi

# Run NJOY with the input file
echo "Running NJOY..."
njoy < "$INPUT" > "$OUTPUT"

# Check if NJOY ran successfully
if [ $? -eq 0 ]; then
    echo "NJOY ran successfully. Output written to $OUTPUT."
else
    echo "NJOY encountered an error. Check $OUTPUT for details."
    exit 1
fi

```

I would appreciate any guidance on this issue! Thank you very much.

**Comment by nathangibson14** [comment, marked as answer]
Hello,

NJOY expects the filenames to be "tape20" and "tape21" with this input.  Your script doesn't appear to rename them, only puts them into an environmental variable.  Hopefully it's as simple as changing those filenames.

--Nathan

**Reply by eitan-weinstein** [reply]
Hi Nathan,

Thanks for responding so quickly to my question. It looks like that was the fix that I needed, so I really appreciate you helping me out!

**Comment by jchsublet** [comment]
Hi,

@eitan-weinstein @nathangibson14 

If founding tape20 and tape21 will indeed allow groupr to run, the above groupr input limits what is available from Fe056 TENDL-2017. To see the full extend of its content using groupr use the automated processing cards

3/
10/
0/
0/

In such a way groupr will output all sections of MF-3, MF-10 up to 30 MeV and also MF-3/mt5*mf6 from 30 to 200 MeV following the ZAP dictionary in MF-8 if the groupr structure selected cover the above ranges
